pkg/auth/audit/filter.go

```diff
@@ -65,14 +73,23 @@ func (h auditHandler) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
 
 	auditLog, err := newAuditLog(h.auditWriter, req, h.sanitizingRegex)
 	if err != nil {
-		util.ReturnHTTPError(rw, req, 500, err.Error())
+		util.ReturnHTTPError(rw, req, http.StatusInternalServerError, err.Error())
 		return
 	}
 
 	wr := &wrapWriter{ResponseWriter: rw, auditWriter: h.auditWriter, statusCode: http.StatusOK}
 	h.next.ServeHTTP(wr, req)
 
-	auditLog.write(user, req.Header, wr.Header(), wr.statusCode, wr.buf.Bytes())
+	err = auditLog.write(user, req.Header, wr.Header(), wr.statusCode, wr.buf.Bytes())
+	if err == nil {
+		return
+	}
+	h.errLock.Lock()
+	defer h.errLock.Unlock()
+	if lastSeen, ok := h.errMap[err.Error()]; !ok || time.Since(lastSeen) > errorDebounceTime {
```

Comment:
Can you add a comment for why this is needed instead of just logging the error directly?

Reply:
Yes, though I am curious about your thoughts around debouncing the error log messages, such as if it is unneeded complexity. The goal is to not spam rancher with failed log messages if the user does something that will cause all logs to fail, such as providing an invalid log path. I think it is nice to have, but not completely necessary. I also just picked an arbitrary debounce time of 30 seconds.